From 5081e737d9ed6ee70230b27953f1eff41feaf268 Mon Sep 17 00:00:00 2001
From: Steven Clark <steven.clark@hashicorp.com>
Date: Mon, 27 Jan 2025 14:34:07 -0500
Subject: [PATCH] Prepare code base for Go 1.24 update. (#29412)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

* Fix "t.Fatal from a non-test goroutine" errors in cache_test.go

 - t.Fatal(f) should not be called within a Go routine based on it's documentation and only from the main test's thread.
 - In 1.24 this seems to cause build failures

* Address all "non-constant format string errors" from go vet

 - Within 1.24 these now cause test builds to fail

…" from go vet

[Backport only api/secret.go to API v1.14.0; other changes affect the
Vault server and SDK, which this package does not ship.]

(cherry picked from commit 9456671f046bae5a4f481088f3ffe5e23d921608)
Signed-off-by: cl2t <yihong.or@isrc.iscas.ac.cn>
---
 api/secret.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/api/secret.go b/api/secret.go
index d37bf3c..a6f59af 100644
--- a/api/secret.go
+++ b/api/secret.go
@@ -6,6 +6,7 @@ package api
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"reflect"
@@ -376,7 +377,7 @@ func ParseSecret(r io.Reader) (*Secret, error) {
 			if err := json.Unmarshal(errBytes, &errStrArray); err != nil {
 				return nil, err
 			}
-			return nil, fmt.Errorf(strings.Join(errStrArray, " "))
+			return nil, errors.New(strings.Join(errStrArray, " "))
 		}
 
 		// if any raw data is present in resp.Body, add it to secret
